### PR TITLE
ARROW-7226: [Python][Doc] Add note re: JSON format support

### PR DESCRIPTION
**Pull request by arw2019**

### CODE REVIEW

**Comment by pitrou** [comment]
Is this useful? The paragraph above contains this sentence:
> a JSON file consists of multiple JSON objects, one per line, representing individual data rows

Why would you expect something else?

**Comment by jorisvandenbossche** [comment]
> Why would you expect something else?

Because this is not a default "JSON" file, and many people will expect a "json reader" to support that. 
I know it is already mentioned in the text, but given this expectation, I think it is fine to have an additional explicit note about it.

File: docs/source/python/json.rst [hunk 1]
```diff
@@ -38,6 +38,9 @@ The features currently offered are the following:
   such as ``my_data.json.gz``)
 * sophisticated type inference (see below)
 
+.. note::
+   Currently only the line-delimited JSON (LDJSON) format is supported.
```
Comment (jorisvandenbossche) [inline docs/source/python/json.rst:42]: I am not sure how common the "LDJSON" abbreviation is. If you google that, I rather get hits for JSON-LD, which is json for linked data. 
Searching for line delimited json gives "newline-delimited JSON (NDJSON)" (http://ndjson.org/) and "JSON Lines" (http://jsonlines.org/). Wiki mentions all three terms, though: https://en.wikipedia.org/wiki/JSON_streaming#Line-delimited_JSON

**Review by jorisvandenbossche — COMMENTED** [review]
Maybe we could also change the first line "Arrow supports reading columnar data from JSON files" to "Arrow supports reading columnar data from line-delimited JSON files" to be explicit directly.

In addition, the docstring of `read_json` could also be updated.

File: docs/source/python/json.rst [hunk 2]
```diff
@@ -38,6 +38,9 @@ The features currently offered are the following:
   such as ``my_data.json.gz``)
 * sophisticated type inference (see below)
 
+.. note::
+   Currently only the line-delimited JSON (JSON-LD) format is supported.
```
Comment (jorisvandenbossche) [inline docs/source/python/json.rst:42]: I wouldn't use JSON-LD here (and the same in the docstring). 

My previous comment might have been worded a bit confusingly, but JSON-LD is the abbreviation for JSON-Linked Data, and searching for LDJSON ended up at pages about JSON-LD (which I raised as a argument to not use LDJSON). Given the variation in abbreviations used, maybe we can just leave out any abbreviation?

Comment (arw2019) [reply docs/source/python/json.rst:42]: That makes sense! I've removed these references to LD-JSON in the new commit

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks @arw2019 !

**Pull request closed without merging**